### PR TITLE
Add Save As RKE Template cluster action

### PR DESCRIPTION
**Pull request by nwmac**

### CODE REVIEW

File: components/PromptModal.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    },
+  },
+
+  methods: {
+    close() {
+      if (!this.opened) {
+        return;
+      }
+
+      this.errors = [];
```
Comment (nwmac) [inline components/PromptModal.vue:47]: Remove this

File: components/PromptModal.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    styles="background-color: var(--nav-bg); border-radius: var(--border-radius); max-height: 100vh;"
+    height="auto"
+    :scrollable="true"
+    @closed="close()"
+  >
+    <component :is="component" v-if="opened && component" :resources="resources" @close="close()" />
+  </modal>
+</template>
+
+<style lang='scss'>
```
Comment (nwmac) [inline components/PromptModal.vue:67]: add scoped

File: components/PromptModal.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  .promptModal-modal {
+    border-radius: var(--border-radius);
+    overflow: scroll;
+    max-height: 100vh;
+    & ::-webkit-scrollbar-corner {
+      background: rgba(0,0,0,0);
```
Comment (nwmac) [inline components/PromptModal.vue:73]: Check

File: components/dialog/SaveAsRKETemplateDialog.vue [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        this.errors = exceptionToErrorsArray(err);
+        buttonDone(false);
+      }
+    }
+  }
+};
+</script>
+
+<template>
+  <Card class="prompt-restore" :show-highlight-border="false">
```
Comment (nwmac) [inline components/dialog/SaveAsRKETemplateDialog.vue:65]: Change class name

File: models/provisioning.cattle.io.cluster.js [hunk 5]
```diff
@@ -45,6 +45,20 @@ export default {
       enabled:    this.$rootGetters['isRancher'],
     });
 
+    const canSaveAsTemplate = this.isRke1 && this.mgmt.status.driver === 'rancherKubernetesEngine' && !this.mgmt.spec.clusterTemplateName;
```
Comment (nwmac) [inline models/provisioning.cattle.io.cluster.js:48]: Check for update link

File: components/dialog/SaveAsRKETemplateDialog.vue [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    },
+  },
+  methods: {
+    close() {
+      this.$emit('close');
+    },
+
+    async apply(buttonDone) {
+      try {
+        await this.$store.dispatch('rancher/request', {
```
Comment (nwmac) [inline components/dialog/SaveAsRKETemplateDialog.vue:44]: Note that the action is not reported on the cluster, so making direct call

**Review by richard-cox — APPROVED** [review]

**Pull request merged**